omni-relayer/src/workers/near.rs

```diff
@@ -0,0 +1,255 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use futures::future::join_all;
+use log::{error, info, warn};
+
+use alloy::rpc::types::Log;
+use ethereum_types::H256;
+use near_primitives::borsh::BorshSerialize;
+use nep141_connector::Nep141Connector;
+use omni_types::{
+    locker_args::ClaimFeeArgs, near_events::Nep141LockerEvent, prover_args::EvmVerifyProofArgs,
+    prover_result::ProofKind, ChainKind,
+};
+
+use crate::{config, utils};
+
+pub async fn sign_transfer(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.near_init_transfer_events.clone(),
+        )
+        .await
+        else {
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                config.redis.sleep_time_after_events_process_secs,
+            ))
+            .await;
+            continue;
+        };
+
+        let mut handlers = Vec::new();
+        while let Some((nonce, event)) = events.next_item().await {
+            if let Ok(event) = serde_json::from_str::<Nep141LockerEvent>(&event) {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        let Nep141LockerEvent::InitTransferEvent { transfer_message } = event
+                        else {
+                            warn!("Expected InitTransferEvent, got: {:?}", event);
+                            return;
+                        };
+
+                        info!(
+                            "Received InitTransferEvent: {}",
+                            transfer_message.origin_nonce.0
+                        );
+
```

Comment:
we also should check fee here.

Reply:
Hmm, I don't think so. I'm checking it right before adding this event to redis in utils/near.rs. The time difference between adding event to db and processing it in worker is at most 10 seconds, so I doubt that we need to check it here again. However, maybe it makes more sense to move my initial check from utils to this worker

Reply:
@olga24912 Let me know where it would be better to move the claim fee check in your opinion. Right now it's being called from utils/near.rs right after we receive log. So there're 3 options here:
1) Leave check in utils module
2) Move check to near worker
3) Have two checks: in utils and in near worker